### PR TITLE
[android] Rename Java package to app.organicmaps

### PR DESCRIPTION
**Pull request by AndrewShkrob**
Closes: #3553

### CODE REVIEW

**Review by rtsisyk — APPROVED** [review]
Thanks for the contribution. I had this patch initially, but decided to wait for further development of mapsme/omim.

**Comment by vng** [comment]
Well, honestly saying I'm beware of this change.
- Could you please show the output for:
git blame android/jni/app/organicmaps/maps/Framework.cpp
- I remember from old MM times that renaming packages broke something with GP updates. But probably it's not actual now ..

File: .clang-format [hunk 1]
```diff
@@ -306,22 +306,22 @@ IncludeCategories:
   - Regex:           '^"pyhelpers/'
```
Comment (rtsisyk) [inline .clang-format:306]: @AndrewShkrob, this patch looks good at glance. However, there are some concerns. Mail.Ru had a lot of time and resources to do such refactoring when MapsWithMe was rebranded to MapsMe. And they didn't. Why? I don't know. I am pretty sure that they tried multiple times. I have heard a belief from some folks that changing Java package name can somehow affect Google Play reviews. It sounds like fairy tales, but who knows? :-) Have you tried such renaming in the past on a production app?

One more issue we can address is to configure `.git-blame-ignore-revs` to improve `git blame`:

https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#ignore-commits-in-the-blame-view

**Comment by AndrewShkrob** [comment]
@vng. [Full git blame for Framework.cpp](https://github.com/organicmaps/organicmaps/files/9812011/Framework_cpp_git_blame.txt)

<details>
  <summary>Only my changes in Framework.cpp</summary>

```
+0200    1) #include "app/organicmaps/maps/Framework.hpp"
+0200    3) #include "app/organicmaps/core/jni_helper.hpp"
+0200    4) #include "app/organicmaps/maps/UserMarkHelper.hpp"
+0200    5) #include "app/organicmaps/opengl/androidoglcontextfactory.hpp"
+0200    6) #include "app/organicmaps/platform/Platform.hpp"
+0200    7) #include "app/organicmaps/util/FeatureIdBuilder.hpp"
+0200    8) #include "app/organicmaps/util/NetworkPolicy.hpp"
+0200    9) #include "app/organicmaps/vulkan/android_vulkan_context_factory.hpp"
+0200  726) //============ GLUE CODE for app.organicmaps.maps.Framework class =============//
+0200  773) Java_app_organicmaps_maps_Framework_nativeGetAddress(JNIEnv * env, jclass clazz, jdouble lat, jdouble lon)
+0200  780) Java_app_organicmaps_maps_Framework_nativeClearApiPoints(JNIEnv * env, jclass clazz)
+0200  786) Java_app_organicmaps_maps_Framework_nativeParseAndSetApiUrl(JNIEnv * env, jclass clazz, jstring url)
+0200  795) Java_app_organicmaps_maps_Framework_nativeGetParsedRoutingData(JNIEnv * env, jclass clazz)
+0200  798)   static jclass const pointClazz = jni::GetGlobalClassRef(env, "app/organicmaps/maps/api/RoutePoint");
+0200  802)   static jclass const routeDataClazz = jni::GetGlobalClassRef(env, "app/organicmaps/maps/api/ParsedRoutingData");
+0200  804)   static jmethodID const routeDataConstructor = jni::GetConstructorID(env, routeDataClazz, "([Lapp/organicmaps/maps/api/RoutePoint;I)V");
+0200  820) Java_app_organicmaps_maps_Framework_nativeGetParsedSearchRequest(JNIEnv * env, jclass clazz)
+0200  823)   static jclass const cl = jni::GetGlobalClassRef(env, "app/organicmaps/maps/api/ParsedSearchRequest");
+0200  832) Java_app_organicmaps_maps_Framework_nativeGetParsedAppName(JNIEnv * env, jclass)
+0200  838) Java_app_organicmaps_maps_Framework_nativeGetParsedCenterLatLon(JNIEnv * env, jclass)
+0200  850) Java_app_organicmaps_maps_Framework_nativePlacePageActivationListener(JNIEnv *env, jclass clazz,
+0200  858)                                                  "(Lapp/organicmaps/maps/widget/placepage/PlacePageData;)V");
+0200  887) Java_app_organicmaps_maps_Framework_nativeRemovePlacePageActivationListener(JNIEnv *env, jclass)
+0200  899) Java_app_organicmaps_maps_Framework_nativeGetGe0Url(JNIEnv * env, jclass, jdouble lat, jdouble lon, jdouble zoomLevel, jstring name)
+0200  908) Java_app_organicmaps_maps_Framework_nativeGetDistanceAndAzimuth(
+0200  915)   static jclass const daClazz = jni::GetGlobalClassRef(env, "app/organicmaps/maps/bookmarks/data/DistanceAndAzimut");
+0200  925) Java_app_organicmaps_maps_Framework_nativeGetDistanceAndAzimuthFromLatLon(
+0200  930)   return Java_app_organicmaps_maps_Framework_nativeGetDistanceAndAzimuth(env, clazz, merX, merY, cLat, cLon, north);
+0200  934) Java_app_organicmaps_maps_Framework_nativeFormatLatLon(JNIEnv * env, jclass, jdouble lat, jdouble lon, int coordsFormat)
+0200  951) Java_app_organicmaps_maps_Framework_nativeFormatAltitude(JNIEnv * env, jclass, jdouble alt)
+0200  958) Java_app_organicmaps_maps_Framework_nativeFormatSpeed(JNIEnv * env, jclass, jdouble speed)
+0200  967) Java_app_organicmaps_maps_Framework_nativeGetOutdatedCountriesString(JNIEnv * env, jclass)
+0200  973) Java_app_organicmaps_maps_Framework_nativeGetOutdatedCountries(JNIEnv * env, jclass)
+0200  987) Java_app_organicmaps_maps_Framework_nativeToDoAfterUpdate(JNIEnv * env, jclass)
+0200  993) Java_app_organicmaps_maps_Framework_nativeIsDataVersionChanged(JNIEnv * env, jclass)
+0200  999) Java_app_organicmaps_maps_Framework_nativeUpdateSavedDataVersion(JNIEnv * env, jclass)
+0200 1006) Java_app_organicmaps_maps_Framework_nativeGetDataVersion(JNIEnv * env, jclass)
+0200 1012) Java_app_organicmaps_maps_Framework_nativeGetDrawScale(JNIEnv * env, jclass)
+0200 1018) Java_app_organicmaps_maps_Framework_nativePokeSearchInViewport(JNIEnv * env, jclass)
+0200 1024) Java_app_organicmaps_maps_Framework_nativeGetScreenRectCenter(JNIEnv * env, jclass)
+0200 1036) Java_app_organicmaps_maps_Framework_nativeShowTrackRect(JNIEnv * env, jclass, jlong track)
+0200 1042) Java_app_organicmaps_maps_Framework_nativeGetBookmarkDir(JNIEnv * env, jclass)
+0200 1048) Java_app_organicmaps_maps_Framework_nativeGetWritableDir(JNIEnv * env, jclass)
+0200 1054) Java_app_organicmaps_maps_Framework_nativeGetSettingsDir(JNIEnv * env, jclass)
+0200 1060) Java_app_organicmaps_maps_Framework_nativeGetDataFileExt(JNIEnv * env, jclass)
+0200 1066) Java_app_organicmaps_maps_Framework_nativeGetMovableFilesExts(JNIEnv * env, jclass)
+0200 1074) Java_app_organicmaps_maps_Framework_nativeGetBookmarksFilesExts(JNIEnv * env, jclass)
+0200 1081) Java_app_organicmaps_maps_Framework_nativeChangeWritableDir(JNIEnv * env, jclass, jstring jNewPath)
+0200 1090) Java_app_organicmaps_maps_Framework_nativeIsRoutingActive(JNIEnv * env, jclass)
+0200 1096) Java_app_organicmaps_maps_Framework_nativeIsRouteBuilding(JNIEnv * env, jclass)
+0200 1102) Java_app_organicmaps_maps_Framework_nativeIsRouteBuilt(JNIEnv * env, jclass)
+0200 1108) Java_app_organicmaps_maps_Framework_nativeCloseRouting(JNIEnv * env, jclass)
+0200 1114) Java_app_organicmaps_maps_Framework_nativeBuildRoute(JNIEnv * env, jclass)
+0200 1120) Java_app_organicmaps_maps_Framework_nativeRemoveRoute(JNIEnv * env, jclass)
+0200 1126) Java_app_organicmaps_maps_Framework_nativeFollowRoute(JNIEnv * env, jclass)
+0200 1132) Java_app_organicmaps_maps_Framework_nativeDisableFollowing(JNIEnv * env, jclass)
+0200 1138) Java_app_organicmaps_maps_Framework_nativeGenerateNotifications(JNIEnv * env, jclass)
+0200 1153) Java_app_organicmaps_maps_Framework_nativeSetSpeedCamManagerMode(JNIEnv * env, jclass, jint mode)
+0200 1160) Java_app_organicmaps_maps_Framework_nativeGetSpeedCamManagerMode(JNIEnv * env, jclass)
+0200 1166) Java_app_organicmaps_maps_Framework_nativeGetRouteFollowingInfo(JNIEnv * env, jclass)
+0200 1177)   static jclass const klass = jni::GetGlobalClassRef(env, "app/organicmaps/maps/routing/RoutingInfo");
+0200 1188)                             "[Lapp/organicmaps/maps/routing/SingleLaneInfo;ZZ)V");
+0200 1194)     static jclass const laneClass = jni::GetGlobalClassRef(env, "app/organicmaps/maps/routing/SingleLaneInfo");
+0200 1230) Java_app_organicmaps_maps_Framework_nativeGenerateRouteAltitudeChartBits(JNIEnv * env, jclass, jint width, jint height, jobject routeAltitudeLimits)
+0200 1307) Java_app_organicmaps_maps_Framework_nativeShowCountry(JNIEnv * env, jclass, jstring countryId, jboolean zoomToDownloadButton)
+0200 1313) Java_app_organicmaps_maps_Framework_nativeSetRoutingListener(JNIEnv * env, jclass, jobject listener)
+0200 1324) Java_app_organicmaps_maps_Framework_nativeSetRouteProgressListener(JNIEnv * env, jclass, jobject listener)
+0200 1332) Java_app_organicmaps_maps_Framework_nativeSetRoutingRecommendationListener(JNIEnv * env, jclass,
+0200 1341) Java_app_organicmaps_maps_Framework_nativeSetRoutingLoadPointsListener(
+0200 1352) Java_app_organicmaps_maps_Framework_nativeDeactivatePopup(JNIEnv * env, jclass)
+0200 1358) Java_app_organicmaps_maps_Framework_nativeSetMapStyle(JNIEnv * env, jclass, jint mapStyle)
+0200 1366) Java_app_organicmaps_maps_Framework_nativeGetMapStyle(JNIEnv * env, jclass)
+0200 1372) Java_app_organicmaps_maps_Framework_nativeMarkMapStyle(JNIEnv * env, jclass, jint mapStyle)
+0200 1380) Java_app_organicmaps_maps_Framework_nativeSetRouter(JNIEnv * env, jclass, jint routerType)
+0200 1396) Java_app_organicmaps_maps_Framework_nativeGetRouter(JNIEnv * env, jclass)
+0200 1402) Java_app_organicmaps_maps_Framework_nativeGetLastUsedRouter(JNIEnv * env, jclass)
+0200 1408) Java_app_organicmaps_maps_Framework_nativeGetBestRouter(JNIEnv * env, jclass,
+0200 1417) Java_app_organicmaps_maps_Framework_nativeAddRoutePoint(JNIEnv * env, jclass, jstring title,
+0200 1435) Java_app_organicmaps_maps_Framework_nativeRemoveRoutePoint(JNIEnv * env, jclass,
+0200 1443) Java_app_organicmaps_maps_Framework_nativeRemoveIntermediateRoutePoints(JNIEnv * env, jclass)
+0200 1449) Java_app_organicmaps_maps_Framework_nativeCouldAddIntermediatePoint(JNIEnv * env, jclass)
+0200 1455) Java_app_organicmaps_maps_Framework_nativeGetRoutePoints(JNIEnv * env, jclass)
+0200 1460)                                    "app/organicmaps/maps/routing/RouteMarkData");
+0200 1484) Java_app_organicmaps_maps_Framework_nativeGetTransitRouteInfo(JNIEnv * env, jclass)
+0200 1489)                                          "app/organicmaps/maps/routing/TransitStepInfo");
+0200 1513)                                                                      "app/organicmaps/maps/routing/TransitRouteInfo");
+0200 1520)                                                                              "[Lapp/organicmaps/maps/routing/TransitStepInfo;)V");
+0200 1532) Java_app_organicmaps_maps_Framework_nativeReloadWorldMaps(JNIEnv * env, jclass)
+0200 1538) Java_app_organicmaps_maps_Framework_nativeIsDayTime(JNIEnv * env, jclass, jlong utcTimeSeconds, jdouble lat, jdouble lon)
+0200 1545) Java_app_organicmaps_maps_Framework_nativeSet3dMode(JNIEnv * env, jclass, jboolean allow, jboolean allowBuildings)
+0200 1555) Java_app_organicmaps_maps_Framework_nativeGet3dMode(JNIEnv * env, jclass, jobject result)
+0200 1571) Java_app_organicmaps_maps_Framework_nativeSetAutoZoomEnabled(JNIEnv * env, jclass, jboolean enabled)
+0200 1579) Java_app_organicmaps_maps_Framework_nativeSetTransitSchemeEnabled(JNIEnv * env, jclass, jboolean enabled)
+0200 1585) Java_app_organicmaps_maps_Framework_nativeIsTransitSchemeEnabled(JNIEnv * env, jclass)
+0200 1591) Java_app_organicmaps_maps_Framework_nativeSetIsolinesLayerEnabled(JNIEnv * env, jclass, jboolean enabled)
+0200 1599) Java_app_organicmaps_maps_Framework_nativeIsIsolinesLayerEnabled(JNIEnv * env, jclass)
+0200 1605) Java_app_organicmaps_maps_Framework_nativeSaveSettingSchemeEnabled(JNIEnv * env, jclass, jboolean enabled)
+0200 1611) Java_app_organicmaps_maps_Framework_nativeGetAutoZoomEnabled(JNIEnv *, jclass)
+0200 1618) Java_app_organicmaps_maps_Framework_nativeZoomToPoint(JNIEnv * env, jclass, jdouble lat, jdouble lon, jint zoom, jboolean animate)
+0200 1624) Java_app_organicmaps_maps_Framework_nativeDeleteBookmarkFromMapObject(JNIEnv * env, jclass)
+0200 1643) Java_app_organicmaps_maps_Framework_nativeTurnOnChoosePositionMode(JNIEnv *, jclass, jboolean isBusiness, jboolean applyPosition)
+0200 1652) Java_app_organicmaps_maps_Framework_nativeTurnOffChoosePositionMode(JNIEnv *, jclass)
+0200 1658) Java_app_organicmaps_maps_Framework_nativeIsInChoosePositionMode(JNIEnv *, jclass)
+0200 1664) Java_app_organicmaps_maps_Framework_nativeIsDownloadedMapAtScreenCenter(JNIEnv *, jclass)
+0200 1671) Java_app_organicmaps_maps_Framework_nativeGetActiveObjectFormattedCuisine(JNIEnv * env, jclass)
+0200 1681) Java_app_organicmaps_maps_Framework_nativeSetVisibleRect(JNIEnv * env, jclass, jint left, jint top, jint right, jint bottom)
+0200 1687) Java_app_organicmaps_maps_Framework_nativeIsRouteFinished(JNIEnv * env, jclass)
+0200 1693) Java_app_organicmaps_maps_Framework_nativeRunFirstLaunchAnimation(JNIEnv * env, jclass)
+0200 1699) Java_app_organicmaps_maps_Framework_nativeOpenRoutePointsTransaction(JNIEnv * env, jclass)
+0200 1705) Java_app_organicmaps_maps_Framework_nativeApplyRoutePointsTransaction(JNIEnv * env, jclass,
+0200 1712) Java_app_organicmaps_maps_Framework_nativeCancelRoutePointsTransaction(JNIEnv * env, jclass,
+0200 1719) Java_app_organicmaps_maps_Framework_nativeInvalidRoutePointsTransactionId(JNIEnv * env, jclass)
+0200 1725) Java_app_organicmaps_maps_Framework_nativeHasSavedRoutePoints(JNIEnv *, jclass)
+0200 1731) Java_app_organicmaps_maps_Framework_nativeLoadRoutePoints(JNIEnv *, jclass)
+0200 1737) Java_app_organicmaps_maps_Framework_nativeSaveRoutePoints(JNIEnv *, jclass)
+0200 1743) Java_app_organicmaps_maps_Framework_nativeDeleteSavedRoutePoints(JNIEnv *, jclass)
+0200 1749) Java_app_organicmaps_maps_Framework_nativeShowFeature(JNIEnv * env, jclass, jobject featureId)
+0200 1758) Java_app_organicmaps_maps_Framework_nativeMakeCrash(JNIEnv *env, jclass type)
+0200 1764) Java_app_organicmaps_maps_Framework_nativeSetPowerManagerFacility(JNIEnv *, jclass,
+0200 1772) Java_app_organicmaps_maps_Framework_nativeGetPowerManagerScheme(JNIEnv *, jclass)
+0200 1778) Java_app_organicmaps_maps_Framework_nativeSetPowerManagerScheme(JNIEnv *, jclass, jint schemeType)
+0200 1784) Java_app_organicmaps_maps_Framework_nativeSetViewportCenter(JNIEnv *, jclass, jdouble lat, jdouble lon, jint zoom)
+0200 1791) Java_app_organicmaps_maps_Framework_nativeStopLocationFollow(JNIEnv *, jclass)
+0200 1797) Java_app_organicmaps_maps_Framework_nativeSetSearchViewport(JNIEnv *, jclass, jdouble lat,
+0200 1807) Java_app_organicmaps_maps_Framework_nativeHasPlacePageInfo(JNIEnv *, jclass)
```
</details>

125 lines mostly in function names. I don't see any problem here.

File: .clang-format (hunk 1 above)
Comment (AndrewShkrob) [reply .clang-format:306]: Unfortunately, I don't have such experience in renaming production apps, but I found some limitations connected with google play [here](https://android-developers.googleblog.com/2011/06/things-that-cannot-change.html)

> Once you publish your application under its manifest package name, this is the unique identity of the application forever more. Switching to a different name results in an entirely new application, one that can’t be installed as an update to the existing application.

> Just as important as the manifest package name is the certificate that application is signed with. The signing certificate represents the author of the application. If you change the certificate an application is signed with, it is now a different application because it comes from a different author. This different application can’t be uploaded to Market as an update to the original application, nor can it be installed onto a device as an update.

> The exact behavior the user sees when installing an application that has changed in one of these two ways is different:

> If the manifest package name has changed, the new application will be installed alongside the old application, so they both co-exist on the user’s device at the same time.

> If the signing certificate changes, trying to install the new application on to the device will fail until the old version is uninstalled.

I don't know if it is our case, because I don't see `package` attribute defined in `<manifest>`. Maybe it is hidden and available only on your side? 

I can configure `.git-blame-ignore-revs`, no problem. But, I think, at first we should decide if this pull request would be accepted.

Anyway, sooner or later, such patch has to be applied 😊

Comment (rtsisyk) [reply .clang-format:306]: >   Once you publish your application under its manifest package name, this is the unique identity of the application forever more. Switching to a different name results in an entirely new application, one that can’t be installed as an update to the existing application.

We don't change the package name in manifest. It should be fine.

> I can configure .git-blame-ignore-revs, no problem. But, I think, at first we should decide if this pull request would be accepted.

Let's try this `git-blame-ignore-revs` and see how it works.

File: android/AndroidManifest.xml [hunk 2]
```diff
@@ -65,7 +65,7 @@
       tools:targetApi="t">
 
     <activity
-      android:name="com.mapswithme.maps.SplashActivity"
+      android:name="app.organicmaps.maps.SplashActivity"
```
Comment (biodranik) [inline android/AndroidManifest.xml:68]: .maps. also can be removed.

File: android/res/drawable/routing_toolbar_button_night.xml [hunk 3]
```diff
@@ -10,4 +10,4 @@
         android:drawable="@drawable/routing_toolbar_button_normal_night"/>
     </selector>
   </item>
-</ripple>
+</ripple>
```
Comment (biodranik) [inline android/res/drawable/routing_toolbar_button_night.xml:13]: Let's create and merge another PR for changes that are not related to renaming.

File: android/src/app/organicmaps/maps/ChartController.java [hunk 4]
```diff
@@ -20,15 +20,15 @@
...
-import com.mapswithme.maps.bookmarks.data.BookmarkManager;
-import com.mapswithme.maps.bookmarks.data.ElevationInfo;
-import com.mapswithme.maps.widget.placepage.AxisValueFormatter;
-import com.mapswithme.maps.widget.placepage.CurrentLocationMarkerView;
-import com.mapswithme.maps.widget.placepage.FloatingMarkerView;
-import com.mapswithme.util.ThemeUtils;
-import com.mapswithme.util.Utils;
+import app.organicmaps.maps.base.Hideable;
+import app.organicmaps.maps.base.Initializable;
+import app.organicmaps.maps.bookmarks.data.BookmarkManager;
```
Comment (biodranik) [inline android/src/app/organicmaps/maps/ChartController.java:25]: There is no need in .maps. here.

File: android/src/app/organicmaps/maps/DownloadResourcesLegacyActivity.java [hunk 5]
```diff
@@ -1,4 +1,4 @@
-package com.mapswithme.maps;
+package app.organicmaps.maps;
```
Comment (biodranik) [inline android/src/app/organicmaps/maps/DownloadResourcesLegacyActivity.java:1]:
```suggestion
package app.organicmaps;
```
This is our current package name. Without .maps at the end.

File: android/src/app/organicmaps/maps/MwmActivity.java [hunk 6]
```diff
@@ -1,4 +1,4 @@
-package com.mapswithme.maps;
+package app.organicmaps.maps;
```
Comment (biodranik) [inline android/src/app/organicmaps/maps/MwmActivity.java:1]: Files like this one can be in app/organicmaps dir. What are the benefits of keeping them in additional maps subdir?

File: tools/python_tests/permissions_test.py [hunk 7]
```diff
@@ -19,11 +19,11 @@
     "uses-permission: android.permission.WAKE_LOCK",
     "uses-permission: android.permission.BATTERY_STATS",
     "uses-permission: com.google.android.c2dm.permission.RECEIVE",
-    "permission: com.mapswithme.maps.pro.permission.C2D_MESSAGE",
-    "uses-permission: com.mapswithme.maps.pro.permission.C2D_MESSAGE",
+    "permission: app.organicmaps.maps.pro.permission.C2D_MESSAGE",
+    "uses-permission: app.organicmaps.maps.pro.permission.C2D_MESSAGE",
```
Comment (biodranik) [inline tools/python_tests/permissions_test.py:23]: Our package name is app.organicmaps, not app.organicmaps.maps.pro

**Review by biodranik — CHANGES_REQUESTED** [review]
If you have enough energy for renaming, let's do it properly:

0. Rebase.
1. Remove unnecessary .maps. suffix/subdirectory to make the code clearer.
2. Check if you can logically organize some files into a related group/subpackage.
3. TEST the app on your device in debug, beta and release modes.

**Comment by AndrewShkrob** [comment]
@rtsisyk @vng I added `.git-blame-ignore-revs` and checked how it works. No changes are visible through git blame now. Everything is good 😊

@biodranik 
0. Did a rebase
1. Removed `.maps.`
2. Didn't do any refactoring. From my pov it looks good, but I can change something if you want
3. Tested fdroid debug, beta and release on my device. Everything works well

There are some scripts in `data` and `tools` folders. They should be reviewed (and removed?) because I think some of them are not used anymore. For example `tools/python_tests/permissions_test.py`.

File: android/AndroidManifest.xml (hunk 2 above)
Comment (rtsisyk) [reply android/AndroidManifest.xml]: > 3. Check if you can logically organize some files into a related group/subpackage.

Could you please elaborate here?

**Review by biodranik — APPROVED** [review]
LGTM, thanks for your contribution!

@vng @rtsisyk let's merge this.

**Comment by biodranik** [comment]
@AndrewShkrob can you please rebase?

**Review by biodranik — APPROVED** [review]
